app/src/main/java/com/infomaniak/swisstransfer/ui/components/SwissTransferTextField.kt

```diff
@@ -0,0 +1,177 @@
+/*
+ * Infomaniak SwissTransfer - Android
+ * Copyright (C) 2024 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.swisstransfer.ui.components
+
+import android.text.Html
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material3.*
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import com.infomaniak.swisstransfer.R
+import com.infomaniak.swisstransfer.ui.images.AppImages.AppIcons
+import com.infomaniak.swisstransfer.ui.images.icons.EyeCrossedOutSmall
+import com.infomaniak.swisstransfer.ui.images.icons.EyeSmall
+import com.infomaniak.swisstransfer.ui.theme.Margin
+import com.infomaniak.swisstransfer.ui.theme.SwissTransferTheme
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+
+@Composable
+fun SwissTransferTextField(
+    modifier: Modifier = Modifier,
+    isRequired: Boolean = true,
+    label: String? = null,
+    initialValue: String = "",
+    minLineNumber: Int = 1,
+    maxLineNumber: Int = if (minLineNumber > 1) Int.MAX_VALUE else 1,
```

Comment:
Deriving the number of max lines from the number of min lines instinctively feels out of place and like it should not be dependent

Reply:
It's actually quite logical when thought about.
Otherwise we need to add another parameter like the `singleLine` one in the original textField
or force to provides parameter, which is not a good thing.

Reply:
If the goal is only to force a single line when none of the parameters are defined, why not simply default minLineNumber to 1 and default maxLineNumber to 1?

Right now, by default, our text component doesn't grow with the text content but suddenly, when I only specify a minLineNumber, it changes its behavior to start growing with the text. That's what feels strange to me

Reply:
because maxLines prevails on minLines
So if I do as you say, then when I want to use it I specify only minLines = 3
The textField will stay a single line
And if we make a multi line TextField, that's because we want a potentially long message so this is dumb to not let it grows.

Reply:
I agree that minLines = 3 giving a 1 line textfield is an issue but this could be fixed with the following.
```suggestion
    minLineNumber: Int = 1,
    maxLineNumber: Int = if (minLineNumber > 1) minLineNumber else 1,
```

The question at stake here is the "is going to grow automatically" property of a text field changing in, from my point of view, an unintuitive manner. To be sure to understand your point, you defend the idea of that property changing automatically to be better than having to consciously specify that you want to field to grow?

With all this talk I'm also wondering, would sticking to the native maxLine and minLine behavior be more simple and intuitive to use?